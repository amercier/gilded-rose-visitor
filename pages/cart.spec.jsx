import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import ConnectedCart, { Cart } from './cart';

const itemsStub = {
  a: { id: 'a', name: 'ITEM A', quality: 10, sellIn: 10, type: 'STANDARD' },
  b: { id: 'b', name: 'ITEM B', quality: 20, sellIn: 20, type: 'STANDARD' },
  c: { id: 'c', name: 'ITEM C', quality: 30, sellIn: 30, type: 'LEGENDARY' },
};

describe('Cart', () => {
  it('renders a list of items', () => {
    const noop = () => {}; // eslint-disable-line require-jsdoc
    const index = shallow(
      <Cart items={Object.values(itemsStub)} onRemoveItem={noop} />,
    );
    expect(index).toMatchSnapshot();
  });
});

describe('Cart (Redux-connected)', () => {
  const mockStore = configureStore();

  it('uses values of items and cart object from the store', () => {
    const store = mockStore({
      itemReducer: { items: itemsStub },
      cartReducer: { cart: ['a', 'b'] },
    });
    const index = mount(
      <Provider store={store}>
        <ConnectedCart />
      </Provider>,
    );
    const indexComponent = index.find(Cart).first();
    expect(indexComponent.props().items).toEqual([itemsStub.a, itemsStub.b]);
  });
});
