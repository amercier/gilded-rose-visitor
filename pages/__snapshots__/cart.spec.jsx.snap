// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders a list of items 1`] = `
<main>
  <h1>
    Your cart
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Item
        </th>
        <th>
          Type
        </th>
        <th>
          Quality
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="a"
      >
        <td>
          <Link
            as="/i/a"
            href="/item?id=a"
          >
            <a>
              ITEM A
            </a>
          </Link>
        </td>
        <td>
          STANDARD
        </td>
        <td>
          10
        </td>
        <td>
          10
        </td>
        <td>
          <button
            onClick={[Function]}
            type="button"
          >
            Remove from cart
          </button>
        </td>
      </tr>
      <tr
        key="b"
      >
        <td>
          <Link
            as="/i/b"
            href="/item?id=b"
          >
            <a>
              ITEM B
            </a>
          </Link>
        </td>
        <td>
          STANDARD
        </td>
        <td>
          20
        </td>
        <td>
          20
        </td>
        <td>
          <button
            onClick={[Function]}
            type="button"
          >
            Remove from cart
          </button>
        </td>
      </tr>
      <tr
        key="c"
      >
        <td>
          <Link
            as="/i/c"
            href="/item?id=c"
          >
            <a>
              ITEM C
            </a>
          </Link>
        </td>
        <td>
          LEGENDARY
        </td>
        <td>
          30
        </td>
        <td>
          30
        </td>
        <td>
          <button
            onClick={[Function]}
            type="button"
          >
            Remove from cart
          </button>
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <th>
          TOTAL
        </th>
        <th />
        <th />
        <th>
          60
        </th>
        <th />
      </tr>
    </tfoot>
  </table>
  <p>
    <button
      type="button"
    >
      Proceed to checkout
    </button>
  </p>
  <p>
    <Link
      href="/"
    >
      <a>
        Go back to shopping.
      </a>
    </Link>
  </p>
</main>
`;
